FrEee/Scripts/AI/Vanguard/MinistryOfInfrastructure.csx

```diff
@@ -0,0 +1,246 @@
+           || faci.HasAbility("Resource Generation - Radioactives")
+           || faci.HasAbility("Point Generation - Research")
+           || faci.HasAbility("Point Generation - Intelligence"))
+           //ensure a little random in the scores, so not always building what is best. 
+        orderby GetScoreForFacility(planet, faci) + RandomHelper.Range(randMin, randMax) descending
+        select faci).FirstOrDefault(); 
+
+            
+    }
+
+    /// <summary>
+    /// Gets the score for the given facility. 
+    /// </summary>
+    /// <param name="planet"></param>
+    /// <param name="facilityTemplate"></param>
+    /// <returns></returns>
+    int GetScoreForFacility(Planet planet, FacilityTemplate facilityTemplate)
+    {
+        int score = 0;
+
+        var prefix = "Resource Generation - ";
+        var pcts = planet.StandardIncomePercentages;
+        foreach (var abil in facilityTemplate.Abilities().Where(abil => abil.Rule.Name.StartsWith(prefix)))
+        {
+            var resourceName = abil.Rule.Name.Substring(prefix.Length);
+            var resource = Resource.Find(resourceName);
+            var amount = abil.Value1.ToInt();
+
+            if (resource.HasValue)
+                amount = Galaxy.Current.StandardMiningModel.GetRate(amount, planet.ResourceValue[resource], pcts[resource] / 100d);
+
+            score += amount + GetResourceScoreMultiplier(resource); 
+        }
+        prefix = "Point Generation - ";
+        foreach (var abil in facilityTemplate.Abilities().Where(abil => abil.Rule.Name.StartsWith(prefix)))
+        {
+            var resoruceName = abil.Rule.Name.Substring(prefix.Length);
+            var resource = Resource.Find(resoruceName);
+            var amount = abil.Value1.ToInt() * pcts[resource] / 100;
+
+            score += amount + GetResourceScoreMultiplier(resource); 
+        }
+
+        return score; 
+    }
+
+    /// <summary>
+    /// Modifies the default score of a facility by the current aspects of the empire. 
+    /// </summary>
+    /// <param name="resourceName"></param>
+    /// <returns></returns>
+    int GetResourceScoreMultiplier(Resource resourceName)
+    {
+        var income = Empire.NetIncomeLessConstruction[resourceName]; 
+        switch(resourceName.Name)
+        {
+            case "Minerals":
+                if (income < 0)
+                    return 10000; 
+                if (income < 10000)
+                    return 1000;
+                if (income < 100000)
+                    return 100;
+                return 0;
+            case "Organics":
+                //TODO: add changes for Organic tech. 
+                if (income < 0)
+                    return 5000;
+                if (income < 10000)
+                    return 500;
+                if (income < 100000)
+                    return 50; 
+                return 0;
+            case "Radioactives":
+                //TODO: add changes for Crystal tech. 
```

Comment:
Also temporal - I guess this AI would only work for stock-like mods? We could analyze available facilities/components to make it generic but is it worth it?

Reply:
For now, I'm aiming for Stock, with others secondary. Get it working in Stock first, then work on making it flexible enough for mods.